join.go

```diff
@@ -0,0 +1,18 @@
+package templ
+
+import (
+	"context"
+	"io"
+)
+
+// Pass any number of templ.Components to get a single templ.Component with the components rendered
```

Comment:
Go function comments should begin with the name of the documented function.

```suggestion
// Join returns a single `templ.Component` that will render provided components in order.
// If any of the components return an error the Join component will immediately return with the error.
```